file-share-service-desktop-client/Modules/Admin/JobViewModels/NewBatchJobViewModel.cs

```diff
@@ -303,7 +343,74 @@ protected internal override async Task OnExecuteCommand()
+            string msg = batchHandle !=null ? string.Format("Are you sure you want to cancel this job batch ID: {0}?", batchHandle.BatchId): "Are you sure you want to cancel this job?";            
+            MessageBoxResult result = MessageBox.Show(msg, "Cancel Job Confirmation", MessageBoxButton.YesNo, MessageBoxImage.Question);
+            if (result == MessageBoxResult.Yes)
+            {               
+                if (CancellationTokenSource != null && !CancellationTokenSource.IsCancellationRequested)
+                {
+                    logger.LogInformation("Cancel job confirmed for Action : {Action}, displayName:{displayName} and batch ID:{BatchId}.", Action, DisplayName, batchHandle?.BatchId);
+                    IsCanceled = true;
+                    IsCommitting = false;                                                               
+                    CancellationTokenSource.Cancel();                                        
+                } 
+            }
+            else
+            {
+                logger.LogInformation("Cancel job canceled for Action : {Action} and displayName :{displayName} and batch ID:{BatchId}.", Action, DisplayName, batchHandle?.BatchId);
+            }
+        }
+
+        private async Task<string> HandleCanceledOperationsAsync(IBatchHandle batchHandle, bool IsCommitting, IFileShareApiAdminClient fileShareClient)
+        {            
+            logger.LogInformation("Cancel job execution started for Action : {Action}, displayName:{displayName} and batch ID:{BatchId}.", Action, DisplayName, batchHandle.BatchId);               
+            if (IsCommitting)
+            {
+                logger.LogInformation("File Share Service SetExpiryDateAsync started for batch ID:{BatchId}.", batchHandle.BatchId);
+                string expiryDateString = DateTime.UtcNow.AddDays(-7).ToString("yyyy-MM-ddTHH:mm:ssZ", CultureInfo.InvariantCulture);
+                await fileShareClient.SetExpiryDateAsync(batchHandle.BatchId, new BatchExpiryModel { ExpiryDate = expiryDateString }, CancellationToken.None);
+                logger.LogInformation("File Share Service SetExpiryDateAsync completed for batch ID:{BatchId}.", batchHandle.BatchId);
+            }
+            else
+            {
+                try
+                {
+                    logger.LogInformation("File Share Service batch rollback started for batch ID:{BatchId}.", batchHandle.BatchId);
+                    await fileShareClient.RollBackBatchAsync(batchHandle);
+                    logger.LogInformation("File Share Service batch rollback completed for batch ID:{BatchId}.", batchHandle.BatchId);
+                }
+                catch (Exception ex)
+                {
+                    if (((System.Net.Http.HttpRequestException)ex).StatusCode.Equals(HttpStatusCode.Conflict))
```

Comment:
It is not safe to cast `ex` to `System.Net.Http.HttpRequestException`. `RollBackBatchAsync` could throw a variety of different exceptions. Please do not make any assumptions about the library

Reply:
Thank you Chris for reviewing this. Made the necessary changes to explicitly handle HttpRequestException with HttpStatusCode.Conflict to adhere to current requirement.